### PR TITLE
fix: app crash when the message links have special characters in it

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

Fixes https://github.com/GetStream/stream-chat-react-native/issues/2297#issuecomment-1813788524

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/src/components/Message/MessageSimple/utils/generateMarkdownText.ts [hunk 1]
```diff
@@ -2,16 +2,11 @@ import truncate from 'lodash/truncate';
...
-
-export const generateMarkdownText = <
-  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
->(
-  message: MessageType<StreamChatGenerics>,
-) => {
-  const { text } = message;
+function escapeRegExp(text: string) {
+  return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+}
```
Comment (santhoshvai) [inline package/src/components/Message/MessageSimple/utils/generateMarkdownText.ts:7]: can you add comments please about what this function does

Comment (santhoshvai) [reply package/src/components/Message/MessageSimple/utils/generateMarkdownText.ts:7]: why does it replace with `\\$&`

its not understandable

Comment (khushal87) [reply package/src/components/Message/MessageSimple/utils/generateMarkdownText.ts:7]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping

Comment (khushal87) [reply package/src/components/Message/MessageSimple/utils/generateMarkdownText.ts:8]: Added comment

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**